python/cuda_parallel/cuda/parallel/experimental/iterators.py

```diff
@@ -0,0 +1,37 @@
+from . import _iterators
+
+
+def CacheModifiedInputIterator(device_array, value_type, modifier):
```

Comment:
question: can we infer value type from device_array? I'd prefer not having `value_type` parameter on this iterator is possible. Value type should match underlying memory's value type exactly.

Reply:
Yes, it should be just `numba.from_dtype(device_array.dtype)`.